### ISSUE TITLE
Add-PodeAuthWindowsAd login form only returns "Username or Password not supplied" 😢

### ISSUE THREAD

**Issue by jackvaughanjr** [Badgerati/Pode#658]
### Question
Windows AD authentication worked in v1.x, but now it doesn't seem to be sending the username/password anywhere? This could be me messing something up. I've included my code below. Perhaps you could let me know where I've messed it up?

In my server.ps1 file:
```powershell
#region Windows Auth
    Enable-PodeSessionMiddleware -Secret 'whatisthissecretevenfor' -Duration 120 -Extend
    
    New-PodeAuthScheme -Form | Add-PodeAuthWindowsAd -Name 'ADLogin' -FailureUrl '/admin/login' -SuccessUrl '/admin/getkey'
    
    Add-PodeRoute -Method Get -Path '/admin/getkey' -Authentication 'ADLogin' -ScriptBlock {
        Write-Host "$($WebEvent.Auth.User.Username) has logged in."
        Write-PodeJsonResponse -Value ($WebEvent.Auth.User)
    }
    
    Add-PodeRoute -Method Get -Path '/admin/login' -Authentication 'ADLogin' -Login -ScriptBlock {
        Write-PodeViewResponse -Path 'auth-login' -FlashMessages
    }
    
    Add-PodeRoute -Method Post -Path '/admin/login' -Authentication 'ADLogin' -Login
```

auth-login.pode
```html
<html lang="en">

<head>
    <!-- Required meta tags -->
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">

    <!-- Bootstrap CSS -->
    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0/css/bootstrap.min.css"
        integrity="sha384-Gn5384xqQ1aoWXA+058RXPxPg6fy4IWvTNh0E263XmFcJlSAwiGgFAW/dAiS6JXm" crossorigin="anonymous">

    <title>AD Authorization Required</title>
</head>

<body>
    <div class="d-md-flex justify-content-center">
        <div class="card">
            <div class="card-header">API Login</div>
            <div class="card-body">
                <form action="/admin/login" method="post">
                    <div class="form-group">
                        <label>Username:</label>
                        <input type="text" class="form-control" name="username" aria-describedby="usernameHelp" placeholder="username" />
                        <small id="usernameHelp" class="form-text text-muted">Please use your AD Creds.</small>
                    </div>
                    <div class="form-group">
                        <label>Password:</label>
                        <input type="password" class="form-control" name="password" placeholder="Password" />
                    </div>
                    <div>
                        <input type="submit" value="Login" class="btn btn-primary" />
                    </div>
                </form>

                $(if ($Data.flash['auth-error']) {
                "<div class='alert alert-danger' role='alert'>$($Data.flash['auth-error'])</div>"
                })
            </div>
        </div>
    </div>
    <!-- Optional JavaScript -->
    <!-- jQuery first, then Popper.js, then Bootstrap JS -->
    <script src="https://code.jquery.com/jquery-3.2.1.slim.min.js"
        integrity="sha384-KJ3o2DKtIkvYIK3UENzmM7KCkRr/rE9/Qpg6aAZGJwFDMVNA/GpGFF93hXpG5KkN"
        crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.12.9/umd/popper.min.js"
        integrity="sha384-ApNbgh9B+Y1QKtv3Rn7W3mgPxhU9K/ScQsAP7hUibX39j7fakFPskvXusvfa0b4Q"
        crossorigin="anonymous"></script>
    <script src="https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0/js/bootstrap.min.js"
        integrity="sha384-JZR6Spejh4U02d8jOt6vLEHfe/JQGiRRSQQxSfFWpi1MquVdAyjUar5+76PVCmYl"
        crossorigin="anonymous"></script>
</body>

</html>
```

**Comment by jackvaughanjr** [comment]
I just tried using only the examples provided in the repo and I get the same response when I submit.

**Comment by Badgerati** [comment]
Hey @jackvaughanjr,

I just tried myself using the example and your code above, and both worked for me 🤔 

Which version of Pode + PowerShell are you using?
Are you hosting behind anything, like IIS?
Is the server using HTTP or HTTPS?

**Comment by jackvaughanjr** [comment]
How strange!

Current Environment:
Pode: 2.0.3
PowerShell: 5.1.14393.3866
HTTPS
Running in a PS window (also experience the same running as a service)

**Comment by jackvaughanjr** [comment]
If I use HTTP, it works as expected.

**Comment by Badgerati** [comment]
I think I may have found the issue. If possible, are you able to test using the code in the `Issue-655` branch? It will require building the .NET listener. This branch contains some logic I'm playing around with to improve the socket reading.

**Comment by jackvaughanjr** [comment]
I would be more than happy to test it out for you, but I would need directions re: building the .NET listener. 🙈

**Comment by RobinBeismann** [comment]
@jackvaughanjr that's pretty easy, download and install the .NET Core SDK (3.1 atm), open a command prompt, navigate into the listener source and type "dotnet build"

**Comment by Badgerati** [comment]
What @RobinBeismann said 😃 

If you have choco you can also do: `choco install dotnetcore --version 3.1.5`.

If you have the InvokeBuild module, you can also navigate to the root of the repo and run `Invoke-Build Build` - this should try to install everything for you.

**Comment by jackvaughanjr** [comment]
I performed the following:

1. `choco install dotnetcore --version 3.1.5`
2. Navigated to code root
3. `Invoke-Build Build`
4. Changed `$ModuleVersion` in Pode.psd1 from `'$version$'` to `2.0.3.675`
5. `Remove-Module -Name Pode` to make sure I didn't have the current release loaded
6. `Import-Module 'c:\pathtolocalrepo\Pode\src\Pode.psd1'`

Then ran my server.ps1 script as I would normally. During the load, I saw a number of 
```PowerShell
WARNING: The names of some imported commands from the module 'PSModule' include unapproved verbs that might make them
less discoverable. To find the commands with unapproved verbs, run the Import-Module command again with the Verbose
parameter. For a list of approved verbs, type Get-Verb.
``` 
warnings, but I assume those aren't an issue as this isn't a "release" version.

Unfortunately, long story long, I receive the same response when attempting to use my AD creds. In the dev console I can see that my username and password are sent in Form Data. 

Is there a step I missed in the build/import process?

**Comment by Badgerati** [comment]
All the steps look correct, just one small question: was this using #675 or #655? I ask because of the `675` on `2.0.3.675` 😛 

The warning are fine, it's just because the PowerShell Gallery module uses unapproved verbs! 😂

**Comment by jackvaughanjr** [comment]
Ah dang. I did use 675. 🙄

Jack Vaughan, Jr.

Systems Monitoring Engineer

P: 314.587.7038<tel:+13145877038> l O: 314.587.7353<tel:+13145877353>

jvaughan@connectria.com<mailto:jvaughan@connectria.com>



________________________________

**Comment by jackvaughanjr** [comment]
This issue is fixed from what I can tell in the 2.1.0 release.

Thank you so much!

**Comment by Badgerati** [comment]
Woop!
